fdbserver/include/fdbserver/TLogInterface.h

```diff
@@ -408,4 +412,36 @@ struct TLogSnapRequest {
 	}
 };
 
+struct TrackTLogRecoveryReply {
+	constexpr static FileIdentifier file_identifier = 5736775;
+
+	// This is the current oldest generation start version this TLog has. This generation is currently being recovered.
+	Version oldestUnrecoveredStartVersion;
+
+	TrackTLogRecoveryReply() = default;
+	explicit TrackTLogRecoveryReply(Version oldestUnrecoveredStartVersion)
+	  : oldestUnrecoveredStartVersion(oldestUnrecoveredStartVersion) {}
+
+	template <class Ar>
+	void serialize(Ar& ar) {
+		serializer(ar, oldestUnrecoveredStartVersion);
+	}
+};
+
+struct TrackTLogRecoveryRequest {
+	constexpr static FileIdentifier file_identifier = 6876454;
+
+	// Reply when the TLog's oldest generation start version is higher than this version.
+	Version oldestGenStartVersion;
+	ReplyPromise<TrackTLogRecoveryReply> reply;
```

Comment:
Since `TrackTLogRecoveryReply` is only about a version:
```suggestion
	ReplyPromise<Version> reply;
```

Reply:
Tried but this doesn't seem to work. It complains that Version doesn't have a serialize function.

Reply:
Yes. `Version` is `int64_t`, which doesn't have the file identifier, which can be solved by adding the following to `flow/serialize.h`:

```
template <>
struct FileIdentifierFor<int64_t> {
	constexpr static FileIdentifier value = 354237;
};
```

I suppose using `TrackTLogRecoveryReply` has the advantage of the unique file identifier. So I don't have a strong opinion here.